base/strings/types.jl

```diff
@@ -122,6 +124,8 @@ reverse(s::RevString) = s.string
 
 isascii(s::RevString{ASCIIString}) = true
 
+Base.convert{T<:AbstractString}(::Type{RevString{T}}, x::SubString{RevString{T}}) = RevString(reverse(bytestring(x)))
```

Comment:
Shouldn't that be `RevString(convert(T, reverse(x)))`?   As written here, it looks like it will always produce a `RevString{T<:ByteString}`.